jooby-neo4j/src/main/java/org/jooby/neo4j/Neo4jSessionStore.java

```diff
@@ -0,0 +1,255 @@
+          Long savedAt = (Long) session.remove("_savedAt");
+          session.remove("_id");
+          session.remove("_expire");
+
+          builder
+                .accessedAt(accessedAt)
+                .createdAt(createdAt)
+                .savedAt(savedAt);
+          session.forEach((k, v) -> builder.set(decode(k), v.toString()));
+          tx.success();
+          return builder.build();
+        }).orElse(null);
+    }
+  }
+
+  @Override
+  public void save(final Session session) {
+    String id = session.id();
+    Map<String, String> attributes = session.attributes();
+
+    try(Transaction tx = db.beginTx()) {
+      Optional.ofNullable(db.findNode(Label.label(sessionLabel), "_id", id))
+        .map(node -> {
+          node.setProperty("_accessedAt", session.accessedAt());
+          node.setProperty("_createdAt", session.createdAt());
+          node.setProperty("_savedAt", session.savedAt());
+
+          attributes.forEach((k ,v) -> node.setProperty(encode(k), v));
+
+          if (!node.hasProperty("_expire")) {
+            node.setProperty("_expire", System.currentTimeMillis() + timeout * 1000);
+          }
+
+          return node;
+        }).orElseGet(() -> {
+          Node node = db.createNode(Label.label(sessionLabel));
+
+          node.setProperty("_accessedAt", session.accessedAt());
+          node.setProperty("_createdAt", session.createdAt());
+          node.setProperty("_savedAt", session.savedAt());
+
+          attributes.forEach((k, v) -> node.setProperty(encode(k), v));
+          node.setProperty("_expire", System.currentTimeMillis() + timeout * 1000);
+
+          return node;
+        });
+      tx.success();
+    }
+  }
+
+  @Override
+  public void create(Session session) {
+    save(session);
+  }
+
+  @Override
+  public void delete(String id) {
+    try(Transaction tx = db.beginTx()) {
+      db.findNode(Label.label(sessionLabel), "_id", id).delete();
+      tx.success();
+    }
+  }
+
+  private static long seconds(final String value) {
+    try {
+      return Long.parseLong(value);
+    } catch (NumberFormatException ex) {
+      Config config = ConfigFactory.empty()
+        .withValue("timeout", ConfigValueFactory.fromAnyRef(value));
+      return config.getDuration("timeout", TimeUnit.SECONDS);
+    } catch (NullPointerException ex) {
+      return 1800L;
+    }
+  }
+
+  private String encode(final String key) {
+    String value = key;
+    if (value.charAt(0) == DOLLAR) {
+      value = UDOLLAR + value.substring(1);
+    }
+    return value.replace(DOT, UDOT);
+  }
+
+  private String decode(final String key) {
+    String value = key;
+    if (value.charAt(0) == UDOLLAR) {
+      value = DOLLAR + value.substring(1);
+    }
+    return value.replace(UDOT, DOT);
+  }
+
+  private static Pair<GraphDatabaseService, GraphAwareRuntime> getOrCreateGraph(String databaseDir) {
+    if (dbService == null && graphRuntime == null) {
```

Comment:
So, in case of ehcache store, an instance of `EhCache` is taken as input. So, do you want to follow that approach & take `dbService` and `graphRuntime` as inputs?